src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -647,11 +673,12 @@ private void UpdateConnectionPair()
 
         _application = applicationToTransport;
         _transport = transportToApplication;
+    }
 
-        // Close previous pipe with specific error that application code can catch to know a restart is occurring
-        prevPipe.Complete(new ConnectionResetException(""));
-
-        Debug.Assert(_notifyOnReconnect is not null);
-        _notifyOnReconnect.Invoke();
+#pragma warning disable CA2252 // This API requires opting into preview features
+    public void DisableReconnect()
+#pragma warning restore CA2252 // This API requires opting into preview features
+    {
+        _useAck = false;
```

Comment:
Doesn't it feel like this should be synchronized? Same with the variant in `HttpConnectionContext`. What if we fire the `notifyOnReconnect` event way after `DisableReconnect()` returns? Potentially after the transport and/or connection have fully closed?